### PR TITLE
Add mapping between user groups and applications.

### PR DESCRIPTION
**Pull request by mdwn**
Within Okta, user groups can assign permission to many applications and applications can belong to many user groups. This commit establishes that mapping so that the Okta service can perform more user friendly access requests by adding all applications from a user group to the access request and additionally allowing application access requests to request associated user groups as well.

### CODE REVIEW

**Review by GavinFrazar — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -5465,6 +5467,18 @@ message UserGroupV1 {
     (gogoproto.jsontag) = "",
     (gogoproto.embed) = true
   ];
+
+  // Spec is the user group resource spec.
+  UserGroupSpecV1 Spec = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
```
[the remark this answers is not in the document]
Comment (mdwn) [reply api/proto/teleport/legacy/types/types.proto:5475]: Ahhh, let me turn nullable to `true` here. Given the lateness of this addition, I think this field has to be optional.

[the remark this answers is not in the document]
Comment (mdwn) [reply api/proto/teleport/legacy/types/types.proto:5475]: Okay, so when trying this out, looks like this behaves as expected as-is with an empty spec (which is what I was hoping would happen):

Before this branch:
```
kind: user_group
metadata:
  description: Humans
  labels:
    okta/org: https://xxx.okta.com
    teleport.dev/origin: okta
    teleport.internal/okta-group-id: xxx
  name: xxx
version: v1
```

Same data set after this branch:

```
kind: user_group
metadata:
  description: Okta Administrators (Okta manages this group, which contains all administrators
    in your organization.)
  labels:
    okta/org: https://xxx.okta.com
    teleport.dev/origin: okta
    teleport.internal/okta-group-id: xxx
  name: xxx
spec: {}
version: v1
```

**Review by jakule — APPROVED** [review]
LGTM

**Comment by mdwn** [comment]
Before this can be checked in, https://github.com/gravitational/teleport.e/pull/1649 will need to be approved so we can do a quick `e` ref update.

**Pull request merged**